1. Reference files

    1. Human Reference Genome - [GRCh38_1000Genomes](http://sourceforge.net/projects/bio-bwa/files/bwakit/bwakit-0.7.11_x64-linux.tar.bz2/download)

    2. Homo_sapiens_assembly38.dbsnp138 https://console.cloud.google.com/storage/browser/genomics-public-data/resources/broad/hg38/v0/Homo_sapiens_assembly38.dbsnp138.vcf

    3. Homo_sapiens_assembly38.known_indels https://console.cloud.google.com/storage/browser/genomics-public-data/resources/broad/hg38/v0/Homo_sapiens_assembly38.known_indels.vcf.gz

    4. Mills_and_1000G_gold_standard.indels.hg38 https://console.cloud.google.com/storage/browser/genomics-public-data/resources/broad/hg38/v0/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz

2. Program Versions

    5. BWA [0.7.15](https://sourceforge.net/projects/bio-bwa/files/bwa-0.7.15.tar.bz2/download)

    6. samtools [1.3.1_htslib_1_3_2](https://sourceforge.net/projects/samtools/)

    7. samblaster [0.1.23](https://github.com/GregoryFaust/samblaster) - change to 0.1.24 (release note on 0.1.23 - THIS RELEASE IS DEPRACATED. PLEASE USE RELEASE 0.1.24 INSTEAD)

    8. sambamba [0.6.4](https://github.com/lomereiter/sambamba/releases)

    9. Picard [2.6.0](https://github.com/broadinstitute/picard/releases/tag/2.6.0)

    10. bamUtil [1.0.13](https://github.com/statgen/bamUtil/releases/tag/v1.0.13)

    11. GATK [3.6](https://software.broadinstitute.org/gatk/download/)

3. Human Resequencing Protocol

    12. Base Stats

        1. Picard Sequence Analyzer

        2. FastQC --disableSeqIDCheck

    13. Mapping

        3. BWA mem -K 100000000 -Y -t 8 -R |

            1. samblaster --addMateTags -a |

            2. samtools view -Sbhu |

            3. sambamba sort -n -t 8

            4. **COMMAND: {bwa_0_7_15} mem -K 100000000 -Y -t 8 -R '{RGstring}' {reference}/reference.fa {FASTQfile1} {FASTQfile2} | {samblaster} --addMateTags -a | {samtools} view -Sbhu - | {sambamba} sort -n -t 8 --tmpdir {TMPDIR} -o {TMPDIR}/{eventId}.tmp.bam /dev/stdin**

        4. picard MarkDuplicates 

                1. ASSUME_SORT_ORDER=queryname 

                2. QUIET=true 

                3. COMPRESSION_LEVEL=0 |

            5. sambamba sort -t 8

            6. **COMMAND: {java_1_8} -jar -Xmx48g {picard} MarkDuplicates I={TMPDIR}/{eventId}.tmp.bam O=/dev/stdout METRICS_FILE={TMPDIR}/mark_dups_metrics.txt ASSUME_SORT_ORDER=queryname QUIET=true COMPRESSION_LEVEL=0 | {sambamba} sort -t 8 --tmpdir {TMPDIR} -o {bamFile} /dev/stdin**

    14. Flagstats

        5. Samtools flagstat

    15. GATK Recal

        6. GenomeAnalysisTK.jar -T BaseRecalibrator

            7. -R GRCh38_1000Genomes                    

            8. -nct 8

            9. --downsample_to_fraction .1

            10. -L chr1 

            11. -L chr2 

            12. -L chr3 

            13. -L chr4 

            14. -L chr5 

            15. -L chr6

            16. -L chr7 

            17. -L chr8 

            18. -L chr9 

            19. -L chr10 

            20. -L chr11 

            21. -L chr12 

            22. -L chr13 

            23. -L chr14 

            24. -L chr15 

            25. -L chr16 

            26. -L chr17 

            27. -L chr18 

            28. -L chr19 

            29. -L chr20 

            30. -L chr21 

            31. -L chr22

            32. -knownSites "Homo_sapiens_assembly38.dbsnp138.vcf"

            33. -knownSites "Mills_and_1000G_gold_standard.indels.hg38.vcf"

            34. -knownSites "Homo_sapiens_assembly38.known_indels.vcf"

            35. **COMMAND: {java_1_8} -Xmx46g -jar {gatk3_6} -T BaseRecalibrator -I {bamFile} -R {reference}/reference.fa -o {recalFile} -nct 8 --downsample_to_fraction .1 -L chr1 -L chr2 -L chr3 -L chr4 -L chr5 -L chr6 -L chr7 -L chr8 -L chr9 -L chr10 -L chr11 -L chr12 -L chr13 -L chr14 -L chr15 -L chr16 -L chr17 -L chr18 -L chr19 -L chr20 -L chr21 -L chr22 -knownSites "Homo_sapiens_assembly38.dbsnp138.vcf" -knownSites "Mills_and_1000G_gold_standard.indels.hg38.vcf" -knownSites "Homo_sapiens_assembly38.known_indels.vcf"**

        7. GenomeAnalysisTK.jar -T -T PrintReads 

            36. -R GRCh38_1000Genomes 

            37. --globalQScorePrior -1.0 

(--useOriginalQualities - removed, doesnâ€™t make a difference)

            38. --preserve_qscores_less_than 6

            39. --static_quantized_quals 10

            40. --static_quantized_quals 20

            41. --static_quantized_quals 30

            42. --disable_indel_quals

            43. **COMMAND: {java_1_8} -Xmx15g -jar {gatk3_6} -T PrintReads -I {bamFile} -R {reference}/reference.fa --BQSR {recalFile} -o {recalBam} --globalQScorePrior -1.0 --preserve_qscores_less_than 6 --static_quantized_quals 10 --static_quantized_quals 20 --static_quantized_quals 30 **

**--disable_indel_quals**

    16. Finishing

        8. samtools view

            44. -C 

            45. -T GRCh38_1000Genomes

            46. -@ 2  

            47. **{samtools} view -C -T {reference}/reference.fa -@ 2 -o {cramFile} {bamFile}**

        9. samtools index

            48. **COMMAND: {samtools} index -c {cramFile} {newIndex}**

